### PR TITLE
Detect better ipfs/ens version

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Continues with #1284

Better detect if the app is in the ENS environment

To detect, it will check for `ipfs` in the domain, or `/ipfs` in the path. This cover all the cases I know :) 

example: https://somehash.ipfs.dweb.link, https://cloudflare-ipfs.com/ipfs/somehash, https://gateway.pinata.cloud/ipfs/code  etc.

before, only `cowswap.eth` was detected, but normally there's a redirection to an IPFS gateway

### CODE REVIEW

**Comment by W3stside** [comment]
does this include the history of https://github.com/gnosis/cowswap/pull/1284 or am i crazy?

**Comment by anxolin** [comment]
> does this include the history of #1284 or am i crazy?

The issue is that i started in the other's PR branch, which was merged. I need to solve the conflicts, and it should be ok after.

File: .env [hunk 1]
```diff
@@ -22,14 +22,17 @@ REACT_APP_NETWORK_URL_100=https://rpc.xdaichain.com
...
 REACT_APP_DOMAIN_REGEX_PR="^pr\d+--gpswapui\.review"
 REACT_APP_DOMAIN_REGEX_DEV="^cowswap\.dev"
 REACT_APP_DOMAIN_REGEX_STAGING="^cowswap\.staging"
 REACT_APP_DOMAIN_REGEX_PROD="^cowswap\.exchange$"
-REACT_APP_DOMAIN_REGEX_ENS="^cowswap\.eth$"
 REACT_APP_DOMAIN_REGEX_BARN="^barn\.cowswap\.exchange$"
+REACT_APP_DOMAIN_REGEX_ENS="(:?^cowswap\.eth$|ipfs)"
+
+# Path regex (to detect environment)
+REACT_APP_PATH_REGEX_ENS="/ipfs"
```
Comment (biocom) [inline .env:35]: What kind of URL's does this detect? Just trying to see if it has the chance to trigger false positives.

**Review by W3stside — APPROVED** [review]

**Review by alfetopito — APPROVED** [review]

File: .env (hunk 1 above)
Comment (anxolin) [reply .env:35]: Urls with `/ipfs` on them. 
I think should be fine as long as we don't include that in the path for any URL.

The other PR with the test has some examples of URL detections.

Comment (anxolin) [reply .env:35]: Merging, happy to address in other PRs if u find any false positive examples

**Pull request merged**